api/src/services/cache/redis-cache.ts

```diff
@@ -0,0 +1,117 @@
+import Redis from 'ioredis';
+import env from '../../env';
+import logger from '../../logger';
+import { getConfigFromEnv } from '../../utils/get-config-from-env';
+import { CacheOptions, CacheService } from './cache';
+import { parse, stringify } from 'json-buffer';
+import { compress, decompress } from '../../utils/compress';
+import { map } from 'async';
+import { merge } from 'lodash';
+
+export class RedisCache extends CacheService {
+	client: Redis;
+
+	constructor(options: CacheOptions) {
+		super(options);
+
+		this.client = new Redis(this.getConfig());
+		this.client.on('error', (err) => logger.warn(err, `[cache] ${err}`));
+		this.client.connect();
+	}
+
+	private getConfig() {
+		if ('CACHE_REDIS' in env) return { url: env.CACHE_REDIS };
```

Comment:
This does not correctly handle connection string config.

**Docs:** https://github.com/luin/ioredis#connect-to-redis

Reply:
Sharp had missed that in the move from `redis` to `ioredis`.